### ISSUE TITLE
Is there a workaround serializing readonly properties?

### ISSUE THREAD

**Issue by nmeri17** [opis/closure#129]
I'm using the version ^3.6 of this library and can upgrade to 4 if this issue is solved there. I have a fairly large codebase where all constructor parameters have the signature `protected readonly type $parameter`. When I try to serialize an object carrying a closure, it says

> error: cannot initialize readonly property classA::property from scope classA

How do I get around this?

**Comment by klkvsk** [comment]
Can you provide a minimal reproducible example?

**Comment by nmeri17** [comment]
Hi, no. I can't. It's been such a long time, I don't recall what part of the project this problem was experienced, but it obviously no longer exists. I may have refactored my implementation not to use this library anymore.

If it haa to do with what I suspect it does, it means I abandoned it altogether on another branch for someone smarter than me to attempt. But you should be able to reproduce with the example I posted. Otherwise, it's OK to close the issue

**Comment by alesinicio** [comment]
The following example triggers the exception brought up by the OP.

```
class Foo {
	public readonly SerializableClosure|Closure $bar;
	public function __construct() {
		$this->bar = function() {
			echo 'Hello world';
		};
	}
}

$wrapped   = \Opis\Closure\serialize(new Foo());
$dewrapped = \Opis\Closure\unserialize($wrapped); //Cannot modify readonly property Foo::$bar
```

**Comment by sorinsarca** [comment]
We just released version 4.0, which supports readonly properties serialization out of the box.
https://opis.io/closure/4.x/
